CONTRIBUTING.MD

```diff
@@ -0,0 +1,135 @@
+
+![Kogito App Resources Composition](docs/img/kogitoapp_resource_composition.png?raw=true)
+
+The `definitions` package ensure that each object is created accordingly. Take for example the `NewRoleBinding` definition:
+
+```go
+func NewRoleBinding(kogitoApp *v1alpha1.KogitoApp, serviceAccount *corev1.ServiceAccount) (roleBinding rbacv1.RoleBinding) {
+	roleBinding = rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			...
+		},
+		RoleRef: rbacv1.RoleRef{
+			...
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      KindServiceAccount.Name,
+				Namespace: serviceAccount.Namespace,
+				Name:      serviceAccount.Name,
+			},
+		},
+    }
+    ...
+	return roleBinding
+}
+```
+
+It will create a new `RoleDefinition` that depends on the `ServiceAccount`, with the default role references, labels and annotations. 
+
+## Contributors Guide
+
+Please follow the sections below to be able to start contributing with the Kogito Operator project.
+
+### Environment
+
+Start with having all the [requirements](README.md#requirements) set on your local machine and **make sure** you stick with the versions described. 
+
+The Operator SDK has regular updates and we will try to catch them as soon as possible. Since they are not on a major version yet, incompatibilities might happen. The same for **dep**.
+
+If you don't have a preferred IDE, we recommend using VS Code with the following plugins:
+
+- [`vscode-go`](https://github.com/Microsoft/vscode-go) for Go lang tools support
+- [`better-toml`](https://github.com/bungcip/better-toml) to be able to read `dep` `*.toml` files.
+
+If you care to debug, follow the ["Debugging Go code using VS Code"](https://github.com/Microsoft/vscode-go/wiki/Debugging-Go-code-using-VS-Code) guide to be able to configure the IDE properly.
+
+### Testing
+
+We love tests and we make our best to always have a unit test to ensure that a certain feature was implemented correctly. 
+
+The Operator SDK has a great documentation explaining how to [unit testing](https://github.com/operator-framework/operator-sdk/blob/master/doc/user/unit-testing.md) an operator. Take your time and have a good read on it to have an overall understanding.
+
+In general, we follow the recommendations listed in that article to write unit tests. There's a caveat that might take you off guard when mocking specific OpenShift APIs like `BuildConfig` and `DeploymentConfig` that are not listed there. Take a look at [`kogitoapp_controller_test.go`](pkg/controller/kogitoapp/kogitoapp_controller_test.go) file as a reference of how to set up your test cases and create fake API calls.
+
+[End-to-end (e2e)](https://github.com/operator-framework/operator-sdk/blob/master/doc/test-framework/writing-e2e-tests.md) tests is a topic that we are pursuing and it's on our [backlog](https://issues.jboss.org/browse/KOGITO-130) to have it solved soon. 
+
+Basically it's an automated integration test written to have it running on a local Kubernetes/OpenShift cluster.
```

Comment:
I do not understand this phrase. To have what "it" running? "To have the entire service running", perhaps?